src/main/java/net/adoptium/documentationservices/model/Documentation.java

```diff
@@ -44,8 +44,25 @@ public Document getDocument(final Locale locale) {
     }
 
     private Document getEnglishDocument() {
-        return getDocuments().filter(d -> Objects.equals(Locale.ENGLISH, d.getLocale()))
+        return getDocuments().filter(d -> Objects.equals(Locale.ENGLISH.getLanguage(), d.getLocale().getLanguage()))
                 .findAny()
                 .orElseThrow(() -> new IllegalStateException("No english document provided for ID=" + id));
     }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final Documentation that = (Documentation) o;
+        return id.equals(that.id);
```

Comment:
There is a check of the documents set missing here. Two Documentation instances should not be considered equals if they have different Documents assigned